gregBorchers/hwWeek1.js

```diff
@@ -0,0 +1,215 @@
+true // x is modified after the comparison
+// **n)** `"1"+x == 1+x`
+false // left side is is a two place string, right side is a number 
+// **o)** `"0"+1 == 1`
+true // wierd, string "01" compares as equal to number 1 must be converting to number?
+// **p)** `(typeof (x+1))===(typeof x)`	
+true // both are numbers (x can be number or string for this to be true)
+
+
+
+// **q)** `(x*1 == x) || ((typeof x) != "number")`
+true // if x is a number, then left side of the OR is true, if x is string then right side of OR is true.  x=NaN returns false
+
+// **r)** `(x=(typeof (x+(typeof x))))==x`
+true // assigning anything to x and then checking to see if x is equivalent to that thing.... seems to always be true.
+
+// ---
+
+// All of the following can be solved with ordinary expressions and global variables with primitive values.  You don't need functions, loops, or other topics beyond our first two classes.
+
+//  **2)** (_Difficulty: easy_)
+
+// Assume variables x, y, and z are numbers.
+
+// **a)**
+// Write an expression for the mean (i.e. average) of x, y, and z.
+
+var mean = (x+y+z)/4;
+
+// **b)**
+// Write a series of expressions to adjust each of x, y, and z halfway toward their mean.
+// That is, reset the value of each variable to something based on its previous value.
+
+x = x + (mean - x) / 2;
+y = y + (mean - y) / 2;
+z = z + (mean - z) / 2;
+
+// ---
+
+// **3)** (_Difficulty: easyish_)
+
+// Suppose you're encoding geometric shapes in a Cartesian (2D) coordinate system, and you represent a rectangle with four numeric variables:
+
+// - _l_ : horizontal position of left edge;
+// - _r_ : horizontal position of right edge;
+// - _t_ : vertical position of top edge;
+// - _b_ : vertical position of bottom edge.
+
+// **a)**
+// Write an expression for the rectangle's area.
+// first make sense of the input variables
+
+var x1 = _l_ 
+var x2 = _r_
+var y1 = _b_
+var y2 = _t_
+
+// now (x1,y1) is lower left corner
+// now (x2,y2) is upper right corner
+var width = x2 - x1;
+var height = y2 - y1;
+
+var area = width * height;
+
+
+// **b)**
+// Write an expression which is true if the rectangle is taller than it is wide, and false otherwise.
+(height>width ? true : false ) 
+
+// **c)**
+// Write an expression for the circumference of the biggest circle which can fit inside the rectangle.  (Hint: you'll need logic similar to that in **b**.)
+// circumference is 2*PI*R
+var maxCircumference = 2 * Math.PI * ((height < width ? height : width )/2) // max circumference is based on radius inscribed inside whichever is smaller height or width
+
+// **d)**
+// Write an expression for the area of the smallest circle which completely encloses (i.e. circumscribes) the rectangle.
+// area is PI*R^2 distance is sqrt( (x2-x1)^2 + (y2-y1)^2 )
+//var minRadius = Math.sqrt( Math.pow((x2-x1),2) + Math.pow((y2-y1),2)) / 2; // half the diagonal is the minimal radius of the circumscribing circle
+var minRadius = (((x2-x1) < y2-y1) ? x2-x1 : y2-y1 );
```

Comment:
This is a good example of why using human-readable variable names is a good idea: your `minRadius` variable assignment has a missing opening paren!